.github/workflows/milestone.yml

```diff
@@ -0,0 +1,61 @@
+name: Check PR milestone
+
+on:
+  pull_request_target:
+    types: [milestoned, demilestoned, opened, reopened]
+    branches:
+      - main
+
+jobs:
+  check_milestone:
+    if: ${{ github.event.issue.pull_request || github.event.pull_request }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # - run: |
+      #     echo "${{ toJSON(github.event) }}"
+
+      - name: "Check for milestone on PR"
+        uses: actions/github-script@v7
+        with:
+          script: |
+            let milestone = context.payload.pull_request.milestone;
+
+            if(context.payload.action === 'opened' || context.payload.action === 'reopened') {
+              if(milestone !== null) {
+                  core.notice(`Milestone is ${milestone.title}`);
+              } else {
+                const milestones = await github.rest.issues.listMilestones({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  state: "open"
+                });
+                for (const milestone of milestones.data) {
```

Comment:
I think `milestone` shadows the previous declaration on l.25. Shouldn't make a difference in this case, but maybe a bit dangerous